crates/sui-graphql-rpc/src/context_data/db_query_cost.rs

```diff
@@ -43,18 +45,38 @@ pub fn raw_sql_string_values_set(
     })?;
     let sql: String = query_builder.finish();
 
+    // handle limits, as '0' is invalid - set to DEFAULT_PAGE_SIZE instead
+    let re = Regex::new(r"(LIMIT\s+)\$(\d+)")
+        .map_err(|e| crate::error::Error::Internal(format!("Failed create valid regex: {}", e)))?;
```

Comment:
- If we need Regexes like this, we should be able to create them once (statically) and then re-use them.
- Setting the limit to `DEFAULT_PAGE_SIZE` kind of defeats the point of this check because it means that someone can create a query that requests 1000x the default page size, and we'll think it costs as much as a default page.  I'm fine with us incrementally building towards the costing algorithm, but do we have a plan for how to get the query with all the parameters in place? Without that, this form of costing is not going to be effective.

Reply:
* ah thanks, I'm not super familiar with regex but will look into that 
* and yes follow up PR to apply the correct bindings. Although on this note, does it make sense for us to limit to default page size or allow users to provide an arbitrary limit? I assumed the former (altho the code doesn't do the capping today)

Reply:
But yes the true cost and estimated cost here are a bit off rn due to replacing the placeholder value with these arbitrary values